clients/samples/gemm/sample_hipblaslt_gemm_epilogue.cpp

```diff
@@ -0,0 +1,146 @@
+                           hipStream_t         stream);
+
+int main()
+{
+    /** This is an example using hipblaslt extension API.
+     *  This is a NN example with epilogue HIPBLASLT_EPILOGUE_BGRADB
+     *  A,B,C,D and Bias are all bf16
+     */
+    Runner<hipblasLtBfloat16, hipblasLtBfloat16, hipblasLtBfloat16, float, float> runner(
+        1024, 512, 1024, 1, 1.f, 1.f, 32 * 1024 * 1024);
+
+    runner.run([&runner] {
+        simpleGemmExtEpilogue(runner.handle,
+                              HIPBLAS_OP_N,
+                              HIPBLAS_OP_T,
+                              HIPBLASLT_EPILOGUE_BGRADB,
+                              runner.m,
+                              runner.n,
+                              runner.k,
+                              runner.batch_count,
+                              runner.alpha,
+                              runner.beta,
+                              runner.d_a,
+                              runner.d_b,
+                              runner.d_c,
+                              runner.d_d,
+                              runner.d_workspace,
+                              runner.max_workspace_size,
+                              runner.stream);
+    });
+
+    return 0;
+}
+
+void simpleGemmExtEpilogue(hipblasLtHandle_t   handle,
+                           hipblasOperation_t  trans_a,
+                           hipblasOperation_t  trans_b,
+                           hipblasLtEpilogue_t epilogue,
+                           int64_t             m,
+                           int64_t             n,
+                           int64_t             k,
+                           int64_t             batch_count,
+                           float&              alpha,
+                           float&              beta,
+                           void*               d_a,
+                           void*               d_b,
+                           void*               d_c,
+                           void*               d_d,
+                           void*               d_workspace,
+                           int64_t             max_workspace_size,
+                           hipStream_t         stream)
+{
+    hipblaslt_ext::GemmPreference gemmPref;
+    gemmPref.setMaxWorkspaceBytes(max_workspace_size);
+    hipblaslt_ext::Gemm gemm(handle,
+                             trans_a,
+                             trans_b,
+                             HIP_R_16BF,
+                             HIP_R_16BF,
+                             HIP_R_16BF,
+                             HIP_R_16BF,
+                             HIPBLAS_COMPUTE_32F);
+
+    hipblaslt_ext::GemmEpilogue gemmEpilogue;
+    gemmEpilogue.mode           = epilogue;
+    gemmEpilogue.bias_data_type = HIP_R_16BF;
+
+    hipblaslt_ext::GemmInputs inputs;
+    inputs.a     = d_a;
+    inputs.b     = d_b;
+    inputs.c     = d_c;
+    inputs.d     = d_d;
+    inputs.alpha = &alpha;
+    inputs.beta  = &beta;
```

Comment:
where is bias pointer??

Reply:
Oh I misses that, but it seems "rocblaslt_epilogue_valid_args()" doesn't catch the error. Let me add the check in that functions too.

Reply:
done